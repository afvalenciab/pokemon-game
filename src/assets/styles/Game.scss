.home--game{
  background-color: rgba(126, 160, 137, .3);
  border-top-left-radius: 30px;
  border-bottom-right-radius: 30px;
  display: grid;
  grid-template-rows: 45px 1fr 45px;
  height: 100%;
  grid-gap: 5px;
  justify-items: center;
  align-items: center;
  box-sizing: border-box;
  box-shadow: 0px 0px 7px -1px rgba(0, 0, 0, 0.5);
  
  h2 {
    margin: 0;
    letter-spacing: 5px;
  }

  .game--item {
    display: flex;
    width: 100%;
    justify-content: space-evenly;
    align-items: center;

    figure {
      margin: 0;
      img {
        vertical-align: top;
        width: 150px;
      }
    }

    .info--game {
      display: flex;
      flex-direction: column;

      .game--life {
        display: flex;
        justify-content: center;
        align-items: center;
  
        .life--bar {
          width: 150px;
          height: 25px;
          box-shadow: 0px 0px 7px -1px rgba(0, 0, 0, 0.5);
          border: 1px solid rgb(114, 113, 113);
          border-radius: 7px;
          background: linear-gradient(90deg, green 100%, #fff 0%);
          margin-right: 10px;
        }

        p {
          width: auto;
          margin: 0;
        }
      }
  
      p {
        width: 100%;
        margin: 0;
        
        .span {
          font-weight: bold;
        }
      }
    }
  }

  button {
    outline: none;
    border: 0;
    font-family: 'Space Mono', monospace;
    background-color: goldenrod;
    width: 300px;
    padding: 7px 15px;
    border-radius: 7px;
    box-shadow: 0px 0px 7px -1px rgba(0, 0, 0, 0.5);
    font-size: 15px;
    letter-spacing: 1.5px;
    cursor: pointer;
    transition: .2s;
    will-change: transform;
  }

  button:hover {
    background-color: rgb(177, 134, 27);
  }
  button:active {
    transform: scale(.92);
  }
}