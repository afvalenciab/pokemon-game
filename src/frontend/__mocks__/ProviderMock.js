import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import initialState from './initialState';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)));
const history = createBrowserHistory();

const ProviderMock = (props) => {
  return(
    <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
    </Provider>
  );
};

export default ProviderMock;