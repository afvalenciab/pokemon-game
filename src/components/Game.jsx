import React, { useState, useContext } from 'react';
import { connect } from 'react-redux';
import { updateLifeBarPokemon } from '../actions';
import { PlayerContext } from '../context/context';
import '../assets/styles/Game.scss';

const Game = props => {
  const { name, picture, lifeBar, moves, updateLifeBarPokemon } = props;
  const player = useContext(PlayerContext);
  const lengthMoves = moves.length;
  const [movesPokemon, setMovesPokemon] = useState({
    move: moves[Math.floor(Math.random()*(lengthMoves - 0))]
  });

  const style = {
    background: `linear-gradient(90deg, green ${lifeBar}%, #fff 0%)`
  };

  const handleClickAttack = (event) => {
    event.preventDefault();
    updateLifeBarPokemon({
      lessLife: (movesPokemon.move.power * 2) / 10,
      player
    });

    const randomPower = Math.floor(Math.random()*(lengthMoves - 0));
    setMovesPokemon({
      move: moves[randomPower]
    });
  };

  return (
    <section className='home--game'>
      <h2>{name}</h2>
      <div className='game--item'>
        <figure>
          <img src={picture} alt='Pokemon' />
        </figure>
        <div className='info--game'>
          <div className='game--life'>
            <div className='life--bar' style={style}></div>
            <p>{lifeBar}%</p>
          </div>
          <p>
            <span className='span'>Ataque: </span> {movesPokemon.move.name}
          </p>
          <p>
            <span className='span'>power:</span> {movesPokemon.move.power}
          </p>
        </div>
      </div>
      <button onClick={handleClickAttack}>Atacar</button>
    </section>
  );
};

const mapDispatchToProps = {
  updateLifeBarPokemon,
};

export default connect(null,mapDispatchToProps)(Game);
