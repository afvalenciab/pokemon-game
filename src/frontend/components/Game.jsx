import React, { useState, useContext } from 'react';
import { connect } from 'react-redux';
import { updateLifeBarPokemon } from '../actions';
import { PlayerContext } from '../context/context';
import '../assets/styles/Game.scss';

const Game = props => {
  const { name, picture, pictureDefault, lifeBar, moves, updateLifeBarPokemon } = props;
  const player = useContext(PlayerContext);
  const lengthMoves = moves ? moves.length : 0;
  const [movesPokemon, setMovesPokemon] = useState({
    move: moves ? moves[Math.floor(Math.random()*(lengthMoves - 0))] : 0
  });

  const [animationImg, setAnimationImg] = useState({
     className: 'img--pokemon'
  });

  const style = {
    background: `linear-gradient(90deg, green ${lifeBar}%, #fff 0%)`
  };

  const handleClickAttack = (event) => {
    event.preventDefault();

    setAnimationImg({
      className: 'img--pokemon movePokemon'
    });

    setTimeout(() => {
      updateLifeBarPokemon({
        lessLife: (movesPokemon.move.power * 2) / 10,
        player
      });
  
      const randomPower = Math.floor(Math.random()*(lengthMoves - 0));
      setMovesPokemon({
        move: moves[randomPower]
      });
      
      setAnimationImg({
        className: 'img--pokemon'
      });
    }, 1000);
  };

  return (
    <section className='home--game'>
      <h2>{name}</h2>
      <div className='game--item'>
        <figure>
          <img key={lifeBar} className={animationImg.className} src={picture ? picture : pictureDefault} alt='Pokemon' />
        </figure>
        <div className='info--game'>
          <div className='game--life'>
            <div className='life--bar' style={style}></div>
            <p>{lifeBar}%</p>
          </div>
          <p>
            <span className='span'>Ataque: </span> {movesPokemon.move.name}
          </p>
          <p>
            <span className='span'>power:</span> {movesPokemon.move.power}
          </p>
        </div>
      </div>
      <button onClick={handleClickAttack}>Atacar</button>
    </section>
  );
};

const mapDispatchToProps = {
  updateLifeBarPokemon,
};

export default connect(null,mapDispatchToProps)(Game);
