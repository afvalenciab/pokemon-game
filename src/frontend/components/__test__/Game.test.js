import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Game from '../Game';

describe('Game Testing', () => {
  test('Match Snapshot', () => {
    const game = create(
      <ProviderMock>
        <Game />
      </ProviderMock>
    );
    expect(game.toJSON()).toMatchSnapshot();
  });
});
