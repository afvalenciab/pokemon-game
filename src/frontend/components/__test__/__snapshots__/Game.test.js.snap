// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Testing Match Snapshot 1`] = `
<section
  className="home--game"
>
  <h2 />
  <div
    className="game--item"
  >
    <figure>
      <img
        alt="Pokemon"
        className="img--pokemon"
      />
    </figure>
    <div
      className="info--game"
    >
      <div
        className="game--life"
      >
        <div
          className="life--bar"
          style={
            Object {
              "background": "linear-gradient(90deg, green undefined%, #fff 0%)",
            }
          }
        />
        <p>
          %
        </p>
      </div>
      <p>
        <span
          className="span"
        >
          Ataque:
           
        </span>
         
      </p>
      <p>
        <span
          className="span"
        >
          Poder:
           
        </span>
         
      </p>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Atacar
  </button>
</section>
`;
